src/ccl_halomod.c

```diff
@@ -181,7 +184,7 @@ static double window_function(ccl_cosmology *cosmo, double m, double k, double a
     rho_matter = ccl_rho_x(cosmo, 1., ccl_species_m_label, 1, status);
 
     // The halo concentration for this mass and scale factor  
-    c = halo_concentration(cosmo, m, a, odelta, Duffy2008_virial, status);    
+    c = ccl_halo_concentration(cosmo, m, a, odelta, Duffy2008_virial, status);    
```

Comment:
Okay, I did a fairly substantial re-write of things to change this. The `halo_concentration` is now elevated to the config for the cosmology object, putting on the same status as `mass_function`. The user can now choose combinations of mass function and halo concentration that they would like to use for the halo-model power-spectrum calculation. The only snag is that the halo-model calculation is fixed to assume that haloes are defined by the virial definition, rather than say M200 or M500. The calculation enforces consistency though, so you *cannot* use a mass function for haloes with M500 in tandem with a concentration relation for haloes defined by M200. For the power spectrum calculation you have to use relations that are defined for virial haloes. I think this is now done!